File: src/main/kotlin/supervinicius/util/FileManager.java
package supervinicius.util;

import java.io.*;
import java.net.URL;

public class FileManager {
    private static FileManager instance;

    private FileManager(){}

    public static FileManager getInstance() {
        if(instance == null) instance = new FileManager();
        return instance;
    }

    public File getFileFromResource(String fileName) {
        ClassLoader classLoader = getClass().getClassLoader();
        URL resource = classLoader.getResource(fileName);

        if(resource == null) {
            throw new IllegalArgumentException("File not found!");
        }
        else return new File(resource.getFile());
    }

    public String getFileContent(File file) {
        if(file == null) return null;

        String result = "";

        try(FileReader reader = new FileReader(file);
            BufferedReader br = new BufferedReader(reader)) {

            String line;

            while((line = br.readLine()) != null) result = result.concat(line + "\n");

        } catch (IOException e) {
            e.printStackTrace();
        }

        return result;
    }

    public boolean writeToFile(String str, String name, String path){
        try {
            File dir = new File(path);
            boolean mkdir = dir.mkdir();
            if(mkdir) Logger.getInstance().debug(path+" dir created!");
            else Logger.getInstance().warn(path+" dir already exist or could not be created!");

            PrintWriter writer = new PrintWriter(new File(path + File.separator + name), "UTF-8");
            writer.println(str);
            writer.close();
            return true;
        }catch (Exception e){
            return false;
        }
    }

    public boolean writeToFile(String str, String name){
        return writeToFile(str, name, "temp");
    }

    public String getFullPath(String str){
        return "temp" + File.separator + str;
    }

    public boolean clearTemp() {
        File folder = new File("temp");
        File[] files = folder.listFiles();
        if(files == null) return true;

        for( File f : files ){
            if(!f.delete()) return false;
        }
        return true;
    }
}

File: src/main/kotlin/leaguebot/util/FileManager.kt
package leaguebot.util

import java.io.*

class FileManager private constructor() {

    @Throws(IOException::class)
    fun getFileFromResource(fileName: String): InputStream {
        val `is` = javaClass.getResourceAsStream(fileName)
        return `is` ?: throw IOException("Failed to open file $fileName")
    }

    fun getFileContent(stream: InputStream?): String {
        if (stream == null) return "null"

        var result = ""

        try {
            BufferedReader(InputStreamReader(stream)).use { br ->
                var line: String

                while (true){
                    line = br.readLine()
                    if(line == null) break

                    result += (line + "\n")
                }

            }
        } catch (e: IOException) {
            Logger.getInstance().error("Failed to load file")
        }

        return result
    }

    @JvmOverloads
    fun writeToFile(str: String, name: String, path: String = "temp"): Boolean {
        try {
            val fullPath: String
            if (path != "") {
                fullPath = path + File.separator + name

                val dir = File(path)
                val mkdir = dir.mkdir()
                if (mkdir)
                    Logger.getInstance().debug("$path dir created!")
                else
                    Logger.getInstance().warn("$path dir already exist or could not be created!")
            } else {
                fullPath = name
            }

            val writer = PrintWriter(File(fullPath), "UTF-8")
            writer.println(str)
            writer.close()
            return true
        } catch (e: Exception) {
            return false
        }

    }

    fun getFullPath(str: String): String {
        return "temp" + File.separator + str
    }

    fun clearTemp(): Boolean {
        val folder = File("temp")
        val files = folder.listFiles() ?: return true

        for (f in files) {
            if (!f.delete()) return false
        }
        return true
    }

    companion object {
        private var instance: FileManager? = null

        fun getInstance(): FileManager {
            if (instance == null) instance = FileManager()
            return instance as FileManager
        }
    }
}
